lib/logstash/outputs/elasticsearch/http_client_builder.rb

```diff
@@ -151,6 +154,18 @@ def self.setup_basic_auth(logger, params)
       }
     end
 
+    def self.setup_api_key(logger, params)
+      api_key = params["api_key"]
+
+      return {} unless (api_key && api_key.value)
+
+      unless params["ssl"] == true
+        raise(LogStash::ConfigurationError, "Using api_key authentication requires SSL/TLS secured communication")
+      end
+
+      { "Authorization" => "ApiKey " + Base64.encode64(api_key.value).chomp }
```

Comment:
`String#chomp` only chomps of the _last_ trailing newline, and will include any intermediate newlines. Instead we can use the RFC4648-compliant [`Base64#strict_encode64`](https://ruby-doc.org/stdlib-2.5.3/libdoc/base64/rdoc/Base64.html#method-i-strict_encode64):

>   Implementations MUST NOT add line feeds to base-encoded data unless
>   the specification referring to this document explicitly directs base
>   encoders to add line feeds after a specific number of characters.
> -- [RFC4648 3.1: Line Feeds In Encoded Data](https://tools.ietf.org/html/rfc4648)

```suggestion
      { "Authorization" => "ApiKey " + Base64.strict_encode64(api_key.value) }
```

Reply:
TIL 👍